### PR TITLE
[chore] protected the user routes and integrated the login api.

### PR DESCRIPTION
**Pull request by Dilum-IR**
# What does this PR do?
Protected the user routes and integrated the login api.

# Primary Reviewer
@buddhi-system2 @NaveenTennakoon

# What have I tested?

### _Network Problem Scenarios_
Verified that a meaningful error message is displayed when the network connection is lost during login.

### _User Routes Protection_
Verified that unauthenticated users attempting to access protected routes (/user/*) are redirected to the login page.

Confirmed that authenticated users can access protected routes after logging in.
Logout and attempt to access a protected route (redirected back to the login page).
Ensured cookies (loggedIn, username) are properly set and removed during login and logout.

## After

# Additional information

### CODE REVIEW

File: src/pages/Login/LoginPage.js [hunk 1]
```diff
@@ -34,32 +35,33 @@ const LoginPage = () => {
...
-            : null;
+        const reqBody = {
+          username,
+          password,
+        };
+        const res = await axios.post(LOGIN_URL, reqBody);
 
-        if (isUser) {
-          createToast("Login Successful!");
+        if (res.status) {
```
Comment (NaveenTennakoon) [inline src/pages/Login/LoginPage.js:52]: We allow logging in if the status is available. Do we not get the status if the login credentials are wrong?

Comment (Dilum-IR) [reply src/pages/Login/LoginPage.js:52]: When login credentials are wrong that response is **bad request(400)** and it handle from the catch block.

### This is the my backend login endpoint
Screenshot 2024-11-20 at 10 12 37

Comment (NaveenTennakoon) [reply src/pages/Login/LoginPage.js:52]: Sure. But the API does return a status in both occasions. Wouldn't this check pass in both occasions therefore?

Comment (Dilum-IR) [reply src/pages/Login/LoginPage.js:52]: I have tested it ayya, but a 400 or other failure statuses causes the try block to not execute the rest of the code.

## Demo Video
https://github.com/user-attachments/assets/e8044c31-0c53-44a9-a431-45cbf5d79714

**Review by NaveenTennakoon — APPROVED** [review]
LGTM!

**Pull request merged**